src/swap/SwapReviewScreen.tsx

```diff
@@ -0,0 +1,374 @@
+import { walletAddressSelector } from 'src/web3/selectors'
+
+type Props = StackScreenProps<StackParamList, Screens.SwapReviewScreen>
+
+const TAG = 'SWAP_REVIEW_SCREEN'
+
+export function SwapReviewScreen(props: Props) {
+  const { toToken, fromToken, swapAmount, updatedField } = props.route.params
+  const [loading, setLoading] = useState(false)
+  const [fetchSwapQuoteError, setFetchSwapQuoteError] = useState(false)
+  const [estimatedModalVisible, setEstimatedDialogVisible] = useState(false)
+  const [swapFeeModalVisible, setSwapFeeModalVisible] = useState(false)
+  const [swapInfo, setSwapInfo] = useState(null as any)
+  const coreTokens = useSelector(coreTokensSelector)
+  const walletAddress = useSelector(walletAddressSelector)
+
+  // Items set for remote config
+  const maxSlippagePercent = useSelector(maxSwapSlippagePercentageSelector)
+  const swapFeeEnabled = useSelector(swapFeeEnabledSelector)
+  const swapFeePercentage = useSelector(swapFeePercentageSelector)
+
+  // Remote configs converted to decimals strings
+  const maxSlippageDecimal = `${maxSlippagePercent / 100}`
+  const swapFeeDecimal = `${swapFeePercentage / 100}`
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  // Token Symbols
+  const toTokenSymbol = coreTokens.find((token) => token.address === toToken)?.symbol
+  const fromTokenSymbol = coreTokens.find((token) => token.address === fromToken)?.symbol
+
+  useEffect(() => {
+    if (fetchSwapQuoteError) {
+      dispatch(showError(ErrorMessages.FETCH_SWAP_QUOTE_FAILED))
+    }
+  }, [fetchSwapQuoteError])
+
+  useEffect(() => {
+    ValoraAnalytics.track(SwapEvents.swap_review_screen_open, {
+      toToken,
+      fromToken,
+      buyAmount: swapAmount[Field.TO] ?? '',
+    })
+  }, [])
+
+  const getSwapInfo = async () => {
+    try {
+      setFetchSwapQuoteError(false)
+      setLoading(true)
+      const swapAmountInWei = multiplyByWei(swapAmount[updatedField]!)
+      const swapAmountParam = updatedField === Field.FROM ? 'sellAmount' : 'buyAmount'
+      const params = {
+        buyToken: toToken,
+        sellToken: fromToken,
+        [swapAmountParam]: swapAmountInWei.toString().split('.')[0],
+        slippagePercentage: maxSlippageDecimal,
+        userAddress: walletAddress,
+      }
+      const queryParams = new URLSearchParams({ ...(params as Record<string, string>) }).toString()
+      const requestUrl = `${networkConfig.approveSwapUrl}?${queryParams}`
+      const response = await fetch(requestUrl)
+      if (!response.ok) {
+        setFetchSwapQuoteError(true)
+        Logger.error(TAG, `Failure response fetching token Swap: ${response}`)
+        throw new Error(
+          `Failure response fetching token swap quote. ${response.status}  ${response.statusText}`
+        )
+      }
+      setSwapInfo(await response.json())
+    } catch (error) {
+      setFetchSwapQuoteError(true)
+      Logger.error(TAG, `Error fetching token swap quote: ${error}`)
```

Comment:
i don't think these should be logger errors right? we only want to surface unhandled or dangerous errors in Sentry, for handled errors like this we shouldn't need it? we'll be aware of the errors from the cloud function side anyway

Reply:
Fixed in https://github.com/valora-inc/wallet/pull/2767/commits/2f5a3f47681caf8a92fc216536f3e63487a8e96c!